tests/rptest/services/databricks_workspace.py

```diff
@@ -0,0 +1,196 @@
+            host=self._databricks_context.server_hostname,
+            token=self._databricks_context.token,
+        )
+
+        # Fail fast if credentials are invalid.
+        self._verify_credentials()
+
+        # Resource trackers.
+        self._location_names: Set[str] = set()
+        self._catalog_names: Set[str] = set()
+
+    def stop(self, **kwargs):
+        self.logger.debug("Stopping Databricks workspace")
+        self._cleanup_catalogs()
+        self._cleanup_dbx_locations()
+        super().stop(**kwargs)
+
+    def create_catalog(self, bucket: str) -> "DatabricksCatalogInfo":
+        """
+        Creates a brand new catalog in the databricks workspace.
+        """
+
+        self._location_names.add(bucket)
+
+        location: ExternalLocationInfo = self._client.external_locations.create(
+            name=bucket,
+            # TODO: Add support for gcs, azure
+            url=f"s3://{bucket}",
+            credential_name=self._databricks_context.credential_name,
+        )
+        self.logger.debug(
+            f"Created external location {location.name} with url {location.url}"
+        )
+        self.logger.debug(f"External location info: {location}")
+
+        requested_catalog_name = f"panda-catalog-{uuid.uuid1()}"
+        self._catalog_names.add(requested_catalog_name)
+
+        catalog_info: CatalogInfo = self._client.catalogs.create(
+            name=requested_catalog_name,
+            storage_root=location.url,
+        )
+        self.logger.debug(f"Created catalog {catalog_info.name}")
+        self.logger.debug(f"Catalog info: {catalog_info}")
+
+        assert catalog_info.catalog_type == CatalogType.MANAGED_CATALOG, \
+            "We expect to only managed catalogs."
+        assert catalog_info.name, "Catalog name must not be empty"
+
+        sql_connection = databricks.sql.connect(
+            server_hostname=self._databricks_context.server_hostname,
+            http_path=self._databricks_context.sql_warehouse_path,
+            access_token=self._databricks_context.token,
+            catalog=catalog_info.name,
+        )
+
+        # This is a unity catalog peculiarity. It allows schemas (iceberg
+        # namespaces) to be created but tables inside it are not allowed
+        # to be created without this grant.
+        #
+        # To avoid this problem we create both, the schema and the grant
+        # before redpanda broker is started.
+        with sql_connection.cursor() as cursor:
+            cursor.execute("CREATE SCHEMA `redpanda`;")
+            self.logger.debug("Schema created successfully")
+
+            principal_row = cursor.execute("SELECT current_user()").fetchone()
+            assert principal_row, "Failed to get current user"
+            principal = principal_row[0]
+
+            self.logger.debug(f"Creating grants for: {principal=}")
+
+            cursor.execute(
+                f"GRANT EXTERNAL USE SCHEMA ON SCHEMA `redpanda` TO `{principal}`"
+            )
```

Comment:
Do you know if the folks from Databricks have plans to do away with this (ie are they working on some hierarchical auth that allows the Schema creator to have the EXTERNAL USE permission by default?)

Reply:
No idea.